.github/workflows/ci-test.yml

```diff
@@ -61,6 +62,12 @@ jobs:
         uses: ./.github/actions/matches
         with:
           pattern: '^.github\|docker/ci\|scripts/dev_setup.sh'
+      - id: docker-compose-changes
+        name: find changes that should trigger docker compose testing.
+        uses: ./.github/actions/matches
+        with:
+          pattern: '^documentation\|^.circleci'
+          invert: "true"
```

Comment:
I would also say this is functionally an integration test between faucet/cli/validator, so there is benefit to running it even when the docker-compose is unchanged

Reply:
agreed, this will only not run when documentation or old circleci changed, the rest of the time it will run, github changes, rust changes, docker changes, shell changes, etc....   I don't expect many people beyond myself to alter this code - or get it right.   Suppose I could more of a write up in the  Readme in the actions folder.